lib/resque/data_store.rb

```diff
@@ -226,26 +235,45 @@ def get_worker_payload(worker_id)
         @redis.get("worker:#{worker_id}")
       end
 
+      def get_worker_queues(worker_id)
+        raw_queues = @redis.get("worker:#{worker_id}:queues")
+        if raw_queues
+          raw_queues.split(',')
+        else
+          []
+        end
```

Comment:
I don't have a strong opinion either way; the explicit `if` is microscopically more efficient (since it avoids calling `''.split(',')`) but that's not particularly relevant.